server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -281,17 +293,38 @@ public void changePassword(final String id, String oldPassword, final String new
         if (oldPassword != null && !checkPasswordMatches(id, oldPassword, zoneId)) {
             throw new BadCredentialsException("Old password is incorrect");
         }
+
+        changePasswordAndWriteHistory(id, newPassword, zoneId);
+    }
+
+    @Override
+    public void changePasswordWithHistoryCheck(final String id, final String oldPassword, final String newPassword, final String zoneId,
+            final int historyLength) throws ScimResourceNotFoundException {
+        if (oldPassword != null && !checkPasswordMatches(id, oldPassword, zoneId)) {
+            throw new BadCredentialsException("Old password is incorrect");
+        }
```

Comment:
So this above block is repeated in `checkPassword()`, suggestions welcome.  Note this is to get the following check sequenced correctly.